types/react-native-appsflyer/index.d.ts

```diff
@@ -0,0 +1,96 @@
+        af_referrer_customer_id?: string;
+        media_source: string;
+        [key: string]: any;
+    };
+}
+
+export enum EmailCryptType {
+    EmailCryptTypeNone = 0,
+    EmailCryptTypeSHA1 = 1,
+    EmailCryptTypeMD5 = 2,
+    EmailCryptTypeSHA256 = 3,
+}
+
+export interface EmailOptions {
+    emailsCryptType: EmailCryptType;
+    emails: string[];
+}
+
+export interface InviteLinkOptions {
+    channel?: string;
+    customerID?: string;
+    campaign?: string;
+    referrerName?: string;
+    referrerImageUrl?: string;
+    deeplinkPath?: string;
+    baseDeeplink?: string;
+    userParams?: {
+        [key: string]: any;
+    };
+}
+
+type SuccessCallback = (success: any) => void;
+
+type ErrorCallback = (error: any) => void;
+
+interface AdditionalData {
+    [key: string]: any;
+}
+
+interface EventValues {
+    [key: string]: any;
+}
+
+declare namespace appsFlyer {
+    function initSdk(options: InitSdkOptions, success?: SuccessCallback, error?: ErrorCallback): void | Promise<string>;
+    function trackAppLaunch(): void;
+    function onInstallConversionData(callback: (data: ConversionData) => void): () => void;
+    function onAppOpenAttribution(callback: (response: any) => void): () => void;
+    function sendDeepLinkData(url: string): void;
+    function stopTracking(isStopTracking: boolean, callback?: SuccessCallback): void;
+    function generateInviteLink(options: InviteLinkOptions, success?: SuccessCallback, error?: ErrorCallback): void;
+    function updateServerUninstallToken(token: string, callback?: SuccessCallback): void;
+    function trackLocation(
+        longitude: number,
+        latitude: number,
+        callback?: (error: any, coordinates: number[]) => void,
+    ): void;
+    function trackCrossPromotionImpression(appId: string, campaign: string): void;
+    function trackAndOpenStore(appId: string, campaign: string, additionalData?: AdditionalData): void;
+    function trackEvent(
+        eventName: string,
+        eventValues: EventValues,
+        success?: SuccessCallback,
+        error?: ErrorCallback,
+    ): void | Promise<string>;
+    function getAppsFlyerUID(callback: (error: any, appsFlyerUID: string) => void): void;
+    function setUserEmails(options: EmailOptions, success?: SuccessCallback, error?: ErrorCallback): void;
+    function setAdditionalData(additionalData: AdditionalData, success?: SuccessCallback): void;
+    function setCustomerUserId(userId: string, callback?: SuccessCallback): void;
+    function setCollectIMEI(isCollect: boolean, callback?: SuccessCallback): void;
+    function setCollectAndroidID(isCollect: boolean, callback?: SuccessCallback): void;
+    function setAppInviteOneLinkID(oneLinkId: string, callback?: SuccessCallback): void;
+    function setCurrencyCode(currencyCode: string, callback?: SuccessCallback): void;
+    function setDeviceTrackingDisabled(isDeviceTrackingDisabled: boolean, success?: SuccessCallback): void;
+}
+
+export default appsFlyer;
```

Comment:
I'm not familiar at all with what exporting a default namespace is supposed to do (I didn't think it was even supported). Can you help me out? Thanks!

Reply:
I'll be honest that I'm not an expert in TS, I referenced other libraries here and came up with writing it this way.
The way I think of it was to follow how the library is implemented, which is just a bunch of methods attached to a JS object. 
So I was trying to do the same with namespace so that it follows how the library APIs are used i.e `appsFlyer.<method>`.
About supporting `export default namespace`, I suppose it is related to this PR.
https://github.com/microsoft/TypeScript/pull/18628

Thanks for reviewing though! I'm learning along the way.